### PR TITLE
[feat][broker]: support dynamic update topic broker-level publish-rate

### PR DESCRIPTION
**Pull request by AnonHxy**
### Motivation

Dynamic update topic broker-level publish-rate.

### Modifications

Invoke `org.apache.pulsar.broker.service.BrokerService#registerConfigurationListener` to register   listeners 
 about `maxPublishRatePerTopicInMessages` and `maxPublishRatePerTopicInBytes` 

### Verifying this change

- [x] Make sure that the change passes the CI checks.
- Add `org.apache.pulsar.broker.service.PrecisTopicPublishRateThrottleTest#testMultiLevelPublishRate` to test the case

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: (no)

### Documentation
  
- [x] `no-need-doc`

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java [hunk 1]
```diff
@@ -1118,4 +1118,9 @@ public void updateBrokerSubscriptionDispatchRate() {
         topicPolicies.getSubscriptionDispatchRate().updateBrokerValue(
             subscriptionDispatchRateInBroker(brokerService.pulsar().getConfiguration()));
     }
+
+    public void updateBrokerPublishRate() {
+        topicPolicies.getPublishRate().updateBrokerValue(
+            publishRateInBroker(brokerService.pulsar().getConfiguration()));
```
Comment (wangjialing218) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java:1124]: When broker level topic publish rate is updated, all topic's publish rate is updated. 
Only those topic which topic level publish rate is null should be updated in this case.

Comment (Jason918) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java:1124]: We don't need to check other level settings with `HierarchyTopicPolicies`

Comment (AnonHxy) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/AbstractTopic.java:1124]: Yes. But we use the `topicPolicies` is a `HierarchyTopicPolicies` type, which will update value through topic-namespace-broker hierarchy. For details : `org.apache.pulsar.common.policies.data.PolicyHierarchyValue#updateBrokerValue` @wangjialing218

**Review by Jason918 — COMMENTED** [review]
Overall LGTM.
Please add unit test for broker level dynamic updating, we missed cover this case.

**Comment by codelipenghui** [comment]
+1 please add test for the new changes.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PrecisTopicPublishRateThrottleTest.java [hunk 2]
```diff
@@ -140,4 +141,64 @@ public void testPrecisTopicPublishRateLimitingProduceRefresh() throws Exception
...
     }
+
+    @Test
+    public void testMultiLevelPublishRate() throws Exception{
+        conf.setPreciseTopicPublishRateLimiterEnable(true);
+        conf.setTopicLevelPoliciesEnabled(true);
+        conf.setSystemTopicEnabled(true);
+        conf.setMaxPendingPublishRequestsPerConnection(0);
+        super.baseSetup();
+        org.apache.pulsar.client.api.Producer<byte[]> producer = pulsarClient.newProducer()
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PrecisTopicPublishRateThrottleTest.java:152]: Close the producer after test.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PrecisTopicPublishRateThrottleTest.java [hunk 3]
```diff
@@ -140,4 +141,64 @@ public void testPrecisTopicPublishRateLimitingProduceRefresh() throws Exception
...
+            .untilAsserted(() ->
+                Assert.assertEquals(admin.brokers().getAllDynamicConfigurations().get("maxPublishRatePerTopicInBytes"),
+                    "" + brokerRate));
+
+        Topic topicRef = pulsar.getBrokerService().getTopicReference(topic).get();
+        Assert.assertNotNull(topicRef);
+        MessageId messageId = null;
+        try {
+            // first will be success, and will set auto read to false
+            messageId = producer.sendAsync(new byte[expectedRate]).get(500, TimeUnit.MILLISECONDS);
```
[the remark this answers is not in the document]
Comment (Jason918) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PrecisTopicPublishRateThrottleTest.java:188]: > I think we only need to check the dynamic update is been applied, do not need to check if the rate limier is works or not?

+1

[the remark this answers is not in the document]
Comment (AnonHxy) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/service/PrecisTopicPublishRateThrottleTest.java]: Have updated the UT, PTAL @codelipenghui @Jason918

**Review by codelipenghui — APPROVED** [review]

**Review by Jason918 — APPROVED** [review]

**Pull request merged**